ops/pebble.py

```diff
@@ -1253,23 +1465,27 @@ class Client:
 
     _chunk_size = 8192
 
-    def __init__(self, socket_path=None, opener=None, base_url='http://localhost', timeout=5.0):
+    def __init__(self, socket_path: str,
+                 opener: Optional[urllib.request.OpenerDirector] = None,
+                 base_url: str = 'http://localhost',
+                 timeout: float = 5.0):
         """Initialize a client instance.
 
         Defaults to using a Unix socket at socket_path (which must be specified
         unless a custom opener is provided).
         """
+        if not isinstance(socket_path, str):
+            raise TypeError('socket_path should be a string; '
+                            'gotten: {}'.format(type(socket_path)))
```

Comment:
Nit: "gotten" reads strangely. Perhaps "socket_path should be a string, not {}"

Reply:
ups, I've been writing that for **ages**, in italian it sounds much better :D

thanks!